### PR TITLE
Lists of hparams are now parsed from dictionaries in yamls

### PR DESCRIPTION
**Pull request by siriuslee**
- Dictionaries should now be used for declaring a list of hparams
- Duplicate entries can be specified using '<field>+<unique-label>', where `<unique-label>' can be anything the user wants

TODO:
- [x] Add docstrings to tests
- [ ] Add documentation on specifying dictionaries for lists
- [ ] Add documentation on duplicate entries

### CODE REVIEW

File: tests/test_yahp_list_from_yaml_dict.py [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+
+
+@pytest.fixture
+def parent_list_hp(data):
+    return ParentListHP.create(data={"foos": data})
+
+
+@pytest.fixture
+def parent_list_hp_no_registry(data):
+    return ParentListHPNoRegistry.create(data=data)
```
Comment (Averylamp) [inline tests/test_yahp_list_from_yaml_dict.py:70]: I don't believe these fixtures are used anywhere?

File: tests/test_yahp_list_from_yaml_dict.py [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+
+def dict_to_list(data: Dict[str, Dict]) -> List[Dict]:
+    """Convert a dictionary to a list of 1-element dictionaries.
+    """
+    return [{k: v} for k, v in data.items()]
+
+
+@pytest.fixture
+def parent_list_hp(data):
+    return ParentListHP.create(data={"foos": data})
```
Comment (Averylamp) [inline tests/test_yahp_list_from_yaml_dict.py:65]: It's pretty dangerous to use data as a fixture variable as that's a very common variable name.  Probably should make it more specific for parameterizing and ensuring no other data fixture is defined

**Review by Averylamp — APPROVED** [review]
LGTM generally, seems like its still failing some tests though, so fix those first

File: tests/test_yahp_list_from_yaml_dict.py (hunk 1 above)
Comment (siriuslee) [reply tests/test_yahp_list_from_yaml_dict.py]: Removed

File: tests/test_yahp_list_from_yaml_dict.py (hunk 2 above)
Comment (siriuslee) [reply tests/test_yahp_list_from_yaml_dict.py]: Removed the fixture. Thanks for the tip, though!

**Pull request merged**